client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,46 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to
+     * 'CONSUMPTION' based retention {@link RetentionPolicy.RetentionType#CONSUMPTION} ,
```

Comment:
I'm wondering about the use of the term `CONSUMPTION` for retention and whether it is going to be confusing to the user. Would terms like `ACKNOWLEDGED`, `CONFIRMED` or `ASSURED` be any better? I don't feel strongly about it at this point, but I want to have a discussion about it.

Reply:
What if instead of adding a new type to the RetentionPolicy enum, we just overloaded the methods in the `StreamConfig`?
Instead of: `TIME`, `SIZE`, and `CONSUMPTION` and the methods `bySizeBytes`, `byTimeMillis`, which take a single value and `byConsumption` which takes two longs and a unit which is either `TIME` or `SIZE` we instead have:
just `TIME`, and `SIZE` and then have `withTimeRangeMillis` and `withSizeRangeBytes` which each take a min and max arg for time/size.
In this case calling `bySizeBytes` is equivalent to passing the same value as the min and the max for `withSizeRangeBytes` (except under the hood we can optimize to do fewer RPCs).
This means that all streams are always use CBR (it's just that some have the same max and min) and so we don't have to explain it as a distinct concept, instead we just say the reader group can request data be retained and when they do so it will be retained provided it is less than the stream's configured max.

Reply:
Filed : https://github.com/pravega/pravega/issues/5356 for this